### ISSUE TITLE
src/consumer-middleware.ts throws error about charset not assignable to parameter of type 'BufferEncoding'

### ISSUE THREAD

**Issue by pleasemarkdarkly** [flitbit/fiver#4]
Building (`yarn build` or `tsc`) at the top level, produces the following error. 

```
src/consumer-middleware.ts:47:53 - error TS2345: Argument of type 'string' is not assignable to parameter of type 'BufferEncoding'.

47         return msg.overrideContent(content.toString(charset));
                                                       ~~~~~~~

src/consumer-middleware.ts:65:51 - error TS2345: Argument of type 'string' is not assignable to parameter of type 'BufferEncoding'.

65           const obj = JSON.parse(content.toString(charset));
                                                     ~~~~~~~
```

Casting the charset like so,

```
    return msg.overrideContent(content.toString(charset as BufferEncoding));
``` 

Makes the build succeed and run, but the worker.ts message output comes through as a byte array versus the JSON of which message was thought to handle. Thoughts, advice, workarounds, pretty please?

**Comment by pleasemarkdarkly** [comment]
My apologies, duh, I can just convert the byte array back to a string on the other side. 
```console.log(new Buffer(m.content, 'base64').toString('utf8'));``` Anyway, the repo produces this error, I'll do a pull request to resolve this, if this solution is reasonable to you.

**Comment by flitbit** [comment]
I'd like to see the PR, in some other projects I've got code that auto-detects and transforms byte arrays back to buffers after serialization to JSON, I like the magic but opinions vary on this; I would ultimately want it to be optional so user decides for themselves whether it is magical or not.

The default should be what it does today, the magic should be opt-in. That way we have "least surprise" and don't break anybody already using the module.

**Comment by pleasemarkdarkly** [comment]
Sorry, just saw this, I've been swamped. I'll do this in a few days when I have "relaxing time."

**Comment by cerebralkungfu** [comment]
Reworked typescript and packaging yesterday. Now outputting a cjs, esm, and browser build. Encountered your error the process. Should be fixed in v1.0.0.
